target/linux/realtek/dts-5.10/rtl9303_hasivo_f1100w-8splus.dts

```diff
@@ -0,0 +1,434 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/dts-v1/;
+
+#include "rtl930x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "hasivo,f1100w-8splus", "realtek,rtl838x-soc";
+	model = "hasivo F1100W-8S+ Switch";
+
+	aliases {
+		led-boot = &led_cpu;
+		led-failsafe = &led_cpu;
+		led-running = &led_cpu;
+		led-upgrade = &led_cpu;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_cpu: led {
+			label = "yellow:cpu";
+			gpios = <&gpio0 21 GPIO_ACTIVE_LOW>;
+		};
+    };
+
+	led_set: led_set@0 {
+		compatible = "realtek,rtl9300-leds";
+		led_set0 = <0xffff 0x0a01 0x0a01 0x0ba0>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	i2c0: i2c-rtl9300 {
+		compatible = "realtek,rtl9300-i2c";
+		reg = <0x1b00036c 0x3c>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		sda-pin = <9>;
+		scl-pin = <8>;
+		clock-frequency = <100000>;
+	};
+
+	i2cmux {
+		compatible = "realtek,i2c-mux-rtl9300";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-parent = <&i2c0>;
+
+		i2c01: i2c-rtl9300-1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <9>;
+			scl-pin = <8>;
+		};
+
+		i2c02: i2c-rtl9300-2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <10>;
+			scl-pin = <8>;
+		};
+
+		i2c03: i2c-rtl9300-3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <11>;
+			scl-pin = <8>;
+		};
+		
+		i2c04: i2c-rtl9300-4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <12>;
+			scl-pin = <8>;
+		};
+
+		i2c05: i2c-rtl9300-5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <13>;
+			scl-pin = <8>;
+		};
+
+		i2c06: i2c-rtl9300-6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <14>;
+			scl-pin = <8>;
+		};
+
+		i2c07: i2c-rtl9300-7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <15>;
+			scl-pin = <8>;
+		};
+	
+		i2c08: i2c-rtl9300-8 {
+			reg = <8>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <16>;
+			scl-pin = <8>;
+		};
+	};
+
+	sfp0: sfp-p1 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c01>;
+		los-gpio = <&gpio0 0 GPIO_ACTIVE_HIGH>;
```

Comment:
You cannot have the same pin for module presence and for LOS. They have different meanings, one says the module is there, the other that the fiber has been unplugged. If you only have one of the 2, then only use the module presence pin.

Reply:
I've noticed it for a while but if I fix it sfp is probed and network goes broken. It is definitely a bug but at this point it'll break everything. So I'm going to have a look your repo and check it works with that.

Reply:
I don't think the kernel can claim 2 gpio's for one function. Do you have any idea how this is electrically connected?

If it is mod-def-0, then it is 'presence detect'.